specification/dartLangSpec.tex

```diff
@@ -10372,12 +10374,50 @@ \subsubsection{Binding Actuals to Formals}
 It is a compile-time error if $q_j = q_k$ for any $j \ne k$.
 
 \LMHash{}%
-If the static type of $f$ is \DYNAMIC{} or the built-in class \FUNCTION{},
-no further static checks are performed and the static type of $i$ is \DYNAMIC{};
-otherwise, it is a compile-time error if the static type of $f$ is not a function type.
+For a given type $T_0$, we introduce the notion of a
+\IndexCustom{$T_0$ bounded type}{type!T0 bounded}:
+$T_0$ itself is $T_0$ bounded;
+if $B$ is $T_0$ bounded and
+$X$ is a type variable with bound $B$
+then $X$ is $T_0$ bounded;
+finally, if $B$ is $T_0$ bounded and
+$X$ is a type variable
+then $X \& B$ is $T_0$ bounded.
+In particular, a
+\IndexCustom{\DYNAMIC{} bounded type}{type!dynamic bounded}
+is either \DYNAMIC{} itself
+or a type variable whose bound is \DYNAMIC{} bounded,
+or an intersection whose second operand is \DYNAMIC{} bounded.
+Similarly for a
+\IndexCustom{\FUNCTION{} bounded type}{type!function bounded}.
```

Comment:
Do we need `Never` bounded types, or does normalization collapse them all to `Never` anyway?
(Are we not handling `Never` in the spec yet?)

Reply:
'Never' doesn't occur in the spec ('Nevertheless' does occur, twice ;). We could add a `TODO` comment about `Never` and the relationship to this topic, but I suspect that the text that we will add about `Never` would include these special cases automatically (for instance, `n.foo` is allowed for all `foo` when `n` has type `Never`, and the expression has type `Never`; that will work for `call` as well).